proxy/common.c

```diff
@@ -36,16 +36,35 @@ validate_namespace (const char * ns)
 }
 
 gboolean
-validate_srvtype (const char * n)
+validate_srvtype(const char *srvtype)
 {
 	gboolean rc = FALSE;
-	NSINFO_READ(if (srvtypes) {
-		for (gchar ** p = srvtypes; !rc && *p; ++p)
-			rc = !strcmp (*p, n);
-	});
+	NSINFO_READ(
+		if (srvtypes) {
+			for (gchar ** p = srvtypes; !rc && *p; ++p)
+				rc = !strcmp(*p, srvtype);
+		});
 	return rc;
 }
 
+GError *
+validate_srvtype_v2(const char *srvtype)
+{
+	gboolean service_types_loaded = FALSE;
+	gboolean service_type_found = FALSE;
+	NSINFO_READ(
+		if (srvtypes) {
+			service_types_loaded = TRUE;
+			for (gchar ** p = srvtypes; !service_type_found && *p; ++p)
+				service_type_found = !strcmp(*p, srvtype);
+		});
+	if (!service_types_loaded)
+		return NEWERROR(CODE_UNAVAILABLE, "Service types not yet loaded");
+	if (!service_type_found)
+		return BADSRVTYPE();
```

Comment:
In the future I think we should modify this macro so it prints the name of the "unexpected" service type.